targets/TARGET_WICED/WICED.md

```diff
@@ -0,0 +1,229 @@
+   ```
+
+1. Change any instances of `_irq` functions to match Mbed OS's linker `_IRQHandler`
+
+   For example:
+   
+   ``` diff
+   - WWD_RTOS_DEFINE_ISR( NoOS_systick_irq )
+   + WWD_RTOS_DEFINE_ISR( NoOS_systick_IRQHandler )
+   ```
+
+1. Comment out the bad `wwd_bus_packet_available_to_read` call while polling for
+   bus packet status.
+
+   In `targets/TARGET_WICED/WICED/WWD/internal/wwd_thread.c` line 336:
+   
+   ``` diff
+     wwd_bus_interrupt = WICED_FALSE;
+
+     /* Check if the interrupt indicated there is a packet to read */
+   - if ( wwd_bus_packet_available_to_read( ) != 0)
+   + // TODO: packet check causes device to lock up^M
+   + if ( /*wwd_bus_packet_available_to_read( )*/ 1 != 0)^M
+     {
+         /* Receive all available packets */
+         do
+   ```
+
+   Note this is required for the WISE1530 but may not be needed for other
+   platforms.
+
+1. Remove the includes of `w_core_cm4.h` and `w_core_cmFunc.h`, and replace the
+   includes with references to Mbed OS's `core_cm4.h`.
+
+   For example:
+   
+   ``` diff
+   - #include "w_core_cmFunc.h"
+   + #include "core_cm4.h"
+   ```
+
+1. Remove the padding of SDPCM data packets. This requires modifications in two
+   places.
+
+   In `targets/TARGET_WICED/WICED/WWD/internal/w_wwd_sdpcm.c` line 210:
+   
+   ``` diff
+    typedef struct
+    {
+        sdpcm_common_header_t  common;
+   -    uint8_t                _padding[2];
+   +    //uint8_t                _padding[2];
+        sdpcm_bdc_header_t     bdc_header;
+    } sdpcm_data_header_t;
+   ```
+
+   In `targets/TARGET_WICED/WICED/WWD/internal/w_wwd_sdpcm.c` line 1282:
+   
+   ``` diff
+        /* Prepare the SDPCM header */
+     memset( (uint8_t*) &packet->sdpcm_header, 0, sizeof(sdpcm_header_t) );
+     packet->sdpcm_header.sw_header.channel_and_flags = (uint8_t) header_type;
+   - packet->sdpcm_header.sw_header.header_length = ( header_type == DATA_HEADER ) ? sizeof(sdpcm_header_t) + 2 : sizeof(sdpcm_header_t);
+   + packet->sdpcm_header.sw_header.header_length = ( header_type == DATA_HEADER ) ? sizeof(sdpcm_header_t) /*+ 2*/ : sizeof(sdpcm_header_t);^M
+     packet->sdpcm_header.sw_header.sequence = 0; /* Note: The real sequence will be written later */
+     packet->sdpcm_header.frametag[0] = size;
+     packet->sdpcm_header.frametag[1] = (uint16_t) ~size;
+   ```
+
+   Note this is required for the WISE1530 but may not be needed for other
+   platforms.
+
+1. Set up SDIO_ENUMERATION_TIMEOUT_MS to be configurable. Mbed OS will override this,
+   depending on the current platform.
+
+   In `targets/TARGET_WICED/WICED/WICED/platform/MCU/STM32F4xx/WWD/wwd_SDIO.c` line 66:
+   
+   ``` diff
+     #define SDIO_IRQ_CHANNEL                     ((u8)0x31)
+     #define DMA2_3_IRQ_CHANNEL                   ((u8)DMA2_Stream3_IRQn)
+     #define BUS_LEVEL_MAX_RETRIES                (5)
+   +
+   + #ifndef SDIO_ENUMERATION_TIMEOUT_MS
+     #define SDIO_ENUMERATION_TIMEOUT_MS          (500)
+   + #endif
+   ```
+
```

Comment:
13. 12 Rename ./mbed-os/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.c to ./mbed-os/targets/TARGET_WICED/TARGET_WISE1530_F412RE/wise1530_emac.cpp

Reply:
Why is this needed? wise1530_emac.c is in Mbed OS, so we can change it here if need be.

But why does it need to be changed to cpp, was there a compilation issue?